src/TreeView/TreeView.tsx

```diff
@@ -22,63 +33,274 @@ export type TreeViewProps = {
   children: React.ReactNode
 }
 
+const UlBox = styled.ul<SxProp>(sx)
+
 const Root: React.FC<TreeViewProps> = ({'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledby, children}) => {
+  const [activeDescendant, setActiveDescendant] = React.useState('')
+
+  React.useEffect(() => {
+    // Initialize the active descendant to the first item in the tree
+    if (!activeDescendant) {
+      const firstItem = document.querySelector('[role="treeitem"]')
+      if (firstItem) setActiveDescendant(firstItem.id)
+    }
+  }, [activeDescendant])
+
   return (
-    <Box
-      as="ul"
-      role="tree"
-      aria-label={ariaLabel}
-      aria-labelledby={ariaLabelledby}
-      sx={{
-        listStyle: 'none',
-        padding: 0,
-        margin: 0
-      }}
-    >
-      {children}
-    </Box>
+    <RootContext.Provider value={{activeDescendant, setActiveDescendant}}>
+      <UlBox
+        tabIndex={0}
+        role="tree"
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabelledby}
+        aria-activedescendant={activeDescendant}
+        sx={{
+          listStyle: 'none',
+          padding: 0,
+          margin: 0,
+          // We'll display a focus ring around the active descendant
+          // instead of the tree itself
+          outline: 0
+        }}
+        onKeyDown={event => {
+          const activeElement = document.getElementById(activeDescendant)
+
+          if (!activeElement) return
+
+          const nextElement = getNextFocusableElement(activeElement, event)
+          if (nextElement) {
+            // Move active descendant if necessary
+            setActiveDescendant(nextElement.id)
+            event.preventDefault()
+          } else {
+            // If the active descendant didn't change,
+            // forward the event to the active descendant
+            activeElement.dispatchEvent(new KeyboardEvent(event.type, event))
+          }
+        }}
+      >
+        {children}
+      </UlBox>
+    </RootContext.Provider>
   )
 }
 
+// DOM utilities used for focus management
```

Comment:
Non-blocking nitpicks: 

1. Would it make more sense to move these to a utilities file instead of keeping them in `TreeView.tsx`?
2. Should these functions have their own unit tests? Or do we think the TreeView tests are sufficient?

Reply:
> 1. Would it make more sense to move these to a utilities file instead of keeping them in TreeView.tsx?

Yeah, it might make sense to pull this out into a hook later. I'm going to keep it all in the same file for now so I can iterate more quickly

> 2. Should these functions have their own unit tests? Or do we think the TreeView tests are sufficient?

I'd like to avoid testing implementation details like this because we'd need to update lots of tests if we decide to change the underlying implementation (even if we don't change behavior). I think the "Keyboard interactions" unit tests give us solid coverage.